diff --git a/volc_conv_ai/src/transports/low_load/src/volc_ws.c b/volc_conv_ai/src/transports/low_load/src/volc_ws.c
index 095a407..64951f9 100644
--- a/volc_conv_ai/src/transports/low_load/src/volc_ws.c
+++ b/volc_conv_ai/src/transports/low_load/src/volc_ws.c
@@ -219,7 +219,7 @@ static int __ws_generate_event_id(ws_impl_t* ws) {
         return -1;
     }
     memset(ws->event_id, 0, sizeof(ws->event_id));
-    snprintf(ws->event_id, sizeof(ws->event_id), "event_id_%llu", ws->event_id_index++);
+    snprintf(ws->event_id, sizeof(ws->event_id), "event_id_%d", ws->event_id_index++);
     return 0;
 }
 
@@ -362,7 +362,7 @@ static int __ws_start(ws_impl_t* ws, volc_iot_info_t* iot_info)
 
     volc_osal_get_platform_info(platform_info, sizeof(platform_info));
     snprintf(user_agent, sizeof(user_agent), "%s(%s)", volc_get_version(), platform_info);
-    snprintf(time_str, sizeof(time_str), "%llu", current_time);
+    snprintf(time_str, sizeof(time_str), "%llu", (unsigned long long)current_time);
     snprintf(num_str, sizeof(num_str), "%d", random_num);
     char* signature = volc_generate_signature_ws(iot_info->device_secret, iot_info->product_key, iot_info->device_name, iot_info->instance_id, random_num, current_time, 1);
     volc_ws_config_t ws_cfg = { 0 };
@@ -468,22 +468,17 @@ static int __ws_response_create(ws_impl_t* ws) {
         LOGE("ws instance is NULL");
         return -1;
     }
+    __ws_generate_event_id(ws);
     root = cJSON_CreateObject();
+    cJSON_AddStringToObject(root, "event_id", ws->event_id);
     cJSON_AddStringToObject(root, "type", "response.create");
-    // 创建 "response" 对象
-    cJSON* response_obj = cJSON_CreateObject();
-    cJSON_AddItemToObject(root, "response", response_obj);
-    // 添加 “modalities” 字段
-    cJSON* modalities_array = cJSON_CreateArray();
-    cJSON_AddItemToArray(modalities_array, cJSON_CreateString("text"));
-    cJSON_AddItemToArray(modalities_array, cJSON_CreateString("audio"));
-    cJSON_AddItemToObject(response_obj, "modalities", modalities_array);
-    // 生成 JSON 字符串
-    const char* json = cJSON_PrintUnformatted(root);
+    char* json = cJSON_PrintUnformatted(root);
     LOGD("json: %s", json);
-    // 释放内存
     cJSON_Delete(root);
     ret = volc_ws_client_send_text(ws->client, json, strlen(json), 1000);
+    if (ret >= 0) {
+        ret = 0;
+    }
     HAL_SAFE_FREE(json);
     return ret;
 }
